### PR TITLE
Fix seats remaining bug with a deepcopy on drivers

### LINKED ISSUE #44: Thursday rides aren't being scheduled
I think Thursday rides aren't being scheduled correctly. Not sure what's going on. Will investigate

### PR DESCRIPTION
**Pull request by rkalnins**
Driver seats_remaining count wasn't reset to original. Fixed with one deepcopy per schedule and then only updating `seats_remaining` every day.

Close #44

### CODE REVIEW

File: scheduler/classes/Day.py [hunk 1]
```diff
@@ -73,3 +73,15 @@ def __init__(self, day: DayName, times: list, locations: list):
...
+
+    def __repr__(self):
+
+        l_str = str()
+        for l in self.locations:
+            l_str = l_str + str(l) + " "
+
+        t_str = str()
+        for t in self.times:
+            t_str = t_str + str(t) + " "
```
Comment (rkalnins) [inline scheduler/classes/Day.py]: @nzsilverman is there a more pythonic way of doing this?

**Comment by rkalnins** [comment]
Rebase diverged so merged instead to update with new changes in develop

File: scheduler/classes/Day.py (hunk 1 above)
Comment (nzsilverman) [reply scheduler/classes/Day.py]: yes, the "proper" way to use a repr is to return a string that is "unambiguous, and, if possible, match the source code necessary to re-create the object being represented." (Fluent Python Book). 

I will push a change that does this

File: scheduler/classes/Day.py [hunk 2]
```diff
@@ -73,3 +73,7 @@ def __init__(self, day: DayName, times: list, locations: list):
         self.day: DayName = day
         self.times: list = times
         self.locations: list = locations
+
+    def __repr__(self):
```
Comment (nzsilverman) [inline scheduler/classes/Day.py:77]: @rkalnins I think my other comment might have been deleted by this push... you asked how to make this more pythonic and this code is my answer to that. The repr should "be unambiguous, and, if possible, match the source code neccesary to re-create the object being represented" (fluent python book).

Comment (nzsilverman) [reply scheduler/classes/Day.py:77]: I also used %r to obtain the standard representation of the attributes that are being shown, so the object could be recreated

**Review by nzsilverman — APPROVED** [review]
looks good, i changed the __repr__ in DayInfo but thats all

**Comment by rkalnins** [comment]
repr string looks really nice, thanks.

**Pull request merged**